### ISSUE TITLE
Use different models for filtering noise and subsequent training

### ISSUE THREAD

**Issue by mm1352363** [cleanlab/cleanlab#253]
Will this affect the performance of the final model? Is it possible to use the same model to have better results?
Looking forward to your answer

**Comment by jwmueller** [comment]
Thanks for your question @mm1352363!
Could you help me understand the specifics a little better? What exactly are the two options you are considering, or what do you hope to do that is not possible in cleanlab v2.0.0?

Note that by default the CleanLearning class will use the same model for both filtering noise and subsequent training, so yes it is possible (and is the default setting). This is our recommended way to do, just define: `clf = YourBestModel()` and then wrap it with: `CleanLearning(clf)`, where you should try to pass in the most accurate model that you can find.

If you prefer to use two different models for filtering vs final training, does this functionality perhaps already handle your use-case?
Since v2.0.0, `CleanLearning.fit()` now has the arguments:
`clf_kwargs` and `clf_final_kwargs`
which allow to you specify one set of kwargs to the models being trained to find & filter label errors and a different set of kwargs to the model being trained at the end. 

https://docs.cleanlab.ai/v2.0.0/cleanlab/classification.html#cleanlab.classification.CleanLearning.fit

Here's a 3rd option you can try if you have 2 completely different models: `model1`, `model2` 
which'll use model1 for filtering and then subsequently train model2 on the filtered clean data.

```
cl1 = CleanLearning(model1)
label_issues = cl1.find_label_issues(data, labels, ...)
cl2 = CleanLearning(model2)
cl2.fit(data, labels, label_issues=label_issues)
```

**Comment by mm1352363** [comment]
@jwmueller thank you for you detail answer, I actually did two  experiments：

1. Using LR in the filtering stage  and NN model (Much more complicated than LR) in subsequent training 
2. Using same NN model in filtering and subsequent training

In the experiment 1, the results are very similar to the results without using cleanlab. And i can get a better result in the experiment 2. 
I would like to know the reason for this result, because using complex model consumes a lot of resource and time in filtering stage.

**Comment by jwmueller** [comment]
Thanks for the details. I assume your NN is much more accurate than your LR? 
If so, that explains why you get a better result in experiment 2.  The more accurate model is much better to use for filtering, because cleanlab's ability to find label issues is dependent on how good your model is.

Here's some tips to improve efficiency you can try out and see if the results are still as good as your experiment 2:

1.  Reduce `cv_n_folds` to say = 3 when you instantiate the `CleanLearning()` object.
2.  Assuming your NN model fit() has some `num_epochs` argument, you can try:
```
CleanLearning.fit(data, labels, clf_kwargs={'num_epochs': A}, clf_final_kwargs = {'num_epochs': B})
```
where `B` = the number of epochs you are currently running, `A` = some smaller number (eg. 10 or 20 say)

**Comment by jwmueller** [comment]
See also: https://github.com/cleanlab/cleanlab/issues/82

Does that satisfy your use-case?

**Comment by jwmueller** [comment]
Closing as I believe this question has been addressed in our new FAQ section:
https://docs.cleanlab.ai/master/tutorials/faq.html#How-can-I-use-different-models-for-data-cleaning-vs.%C2%A0final-training-in-CleanLearning?

Feel free to reopen if you still have questions!
